apps/vaults/hooks/useSolverCowswap.tsx

```diff
@@ -0,0 +1,300 @@
+			quote,
+			signer,
+			SigningScheme.EIP712
+		);
+		return ethers.utils.joinSignature(rawSignature.data);
+	}, [provider, shouldUsePresign, address]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** refreshQuote can be called by the user to refresh the quote. The same parameters are used
+	** as in the initial request and it will fails if request is not set.
+	** init should be called first to initialize the request.current.
+	**********************************************************************************************/
+	const	refreshQuote = useCallback(async (): Promise<void> => {
+		if (request.current) {
+			getQuote(request.current);
+		}
+	}, [request, getQuote]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** approve is a method that approves an order for the CowSwap exchange. It returns a boolean
+	** value indicating whether the approval was successful or not.
+	** This method tries to sign the quote object contained within the latestQuote.result object using
+	** the signCowswapOrder method described above. The buyAmount field of the quote object is set
+	** to the value of the buyAmountWithSlippage variable before it is passed to the
+	** signCowswapOrder method.
+	**********************************************************************************************/
+	const	approve = useCallback(async (): Promise<boolean> => {
+		if (!latestQuote?.current || !latestQuote?.current?.quote || !request?.current) {
+			return false;
+		}
+
+		const	{quote} = latestQuote.current;
+		try {
+			const	buyAmountWithSlippage = getBuyAmountWithSlippage(latestQuote.current, request.current.outputToken.decimals);
+			const	_signature = await signCowswapOrder({
+				...quote,
+				buyAmount: buyAmountWithSlippage.toString()
+			});
+			signature.current = _signature;
+			return true;
+		} catch (_error) {
+			console.error(_error);
+			return false;
+		}
+	}, [latestQuote, signCowswapOrder]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** Cowswap orders have a validity period and the return value on submit is not the execution
+	** status of the order. This method is used to check the status of the order and returns a
+	** boolean value indicating whether the order was successful or not.
+	** It will timeout once the order is no longer valid or after 50 minutes (max should be 30mn)
+	**********************************************************************************************/
+	async function checkOrderStatus(orderUID: string, validTo: Timestamp): Promise<{isSuccessful: boolean, error: Error | undefined}> {
+		for (let i = 0; i < maxIterations; i++) {
+			const {data: order} = await axios.get(`https://api.cow.fi/mainnet/api/v1/orders/${orderUID}`);
+			if (order?.status === 'fulfilled') {
+				return ({isSuccessful: true, error: undefined});
```

Comment:
nit: Unnecessary to explicitly make it `undefined` as it's already by default
```suggestion
				return ({isSuccessful: true});
```